docs/docs/developer/frontend/style-guide.mdx

```diff
@@ -259,3 +259,32 @@ const StyledButton = styled.button`
   margin-left: ${({ theme }) => theme.spacing(1)};
   border-radius:  ${({ theme }) => theme.border.rounded};
 `;
+```
+## Enforcing No-Type Imports
+
+In our codebase, we've adopted a coding standard to disallow type imports. This helps maintain consistency and readability in our TypeScript code. To enforce this standard, we've added an ESLint rule that checks for and reports any type imports.
+
+```tsx
+// ❌ Bad
+import { type Meta, type StoryObj } from '@storybook/react';
+
+// ❌ Bad
+import type { Meta, StoryObj } from '@storybook/react';
+
+// ✅ Good
+import { Meta, StoryObj } from '@storybook/react';
+```
+
+### Why No-Type Imports?
+
+- **Consistency**: By avoiding type imports, our codebase remains consistent in its module import style. We use a single approach for both type and value imports.
+
+- **Readability**: No-type imports improve code readability by making it clear when you're importing values or types. This reduces ambiguity and makes it easier to understand the purpose of imported symbols.
+
+- **Maintainability**: Codebase maintainability is enhanced because developers can quickly identify and locate type-only imports when reviewing or modifying code.
```

Comment:
Could you please add a reference to the documentation which explains it is meant only for rare edge cases : https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html